flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/CsvFormatITCase.java

```diff
@@ -0,0 +1,204 @@
+
+        String jsonPlan =
+                tableEnv.getJsonPlan("insert into MySink select * from MyTable where p = 2");
+        tableEnv.executeJsonPlan(jsonPlan).await();
+
+        assertResult(Arrays.asList("2,2,Hello", "3,2,Hello world"), sinkPath);
+    }
+
+    @Test
+    public void testWatermarkPushDown() throws Exception {
+        createTestValuesSourceTable(
+                "MyTable",
+                JavaScalaConversionUtil.toJava(TestData.data3WithTimestamp()),
+                new String[] {
+                    "a int",
+                    "b bigint",
+                    "c varchar",
+                    "ts timestamp(3)",
+                    "watermark for ts as ts - interval '5' second"
+                },
+                new HashMap<String, String>() {
+                    {
+                        put("enable-watermark-push-down", "true");
+                    }
+                });
+
+        File sinkPath = createSinkTable("MySink", "a int", "b bigint", "ts timestamp(3)");
+
+        String jsonPlan =
+                tableEnv.getJsonPlan("insert into MySink select a, b, ts from MyTable where b = 3");
+        tableEnv.executeJsonPlan(jsonPlan).await();
+
+        assertResult(
+                Arrays.asList(
+                        "4,3," + formatSqlTimestamp(4000L),
+                        "5,3," + formatSqlTimestamp(5000L),
+                        "6,3," + formatSqlTimestamp(6000L)),
+                sinkPath);
+    }
+
+    @Test
+    public void testPushDowns() throws Exception {
+        createTestValuesSourceTable(
+                "MyTable",
+                JavaScalaConversionUtil.toJava(TestData.data3WithTimestamp()),
+                new String[] {
+                    "a int",
+                    "b bigint",
+                    "c varchar",
+                    "ts timestamp(3)",
+                    "watermark for ts as ts - interval '5' second"
+                },
+                "b",
+                new HashMap<String, String>() {
+                    {
+                        put("readable-metadata", "a:INT");
+                        put("filterable-fields", "a");
+                        put("enable-watermark-push-down", "true");
+                        put("partition-list", "b:1;b:2;b:3;b:4;b:5;b:6");
+                    }
+                });
+
+        File sinkPath = createSinkTable("MySink", "a int", "ts timestamp(3)");
+
+        String jsonPlan =
+                tableEnv.getJsonPlan(
+                        "insert into MySink select a, ts from MyTable where b = 3 and a > 4");
+        tableEnv.executeJsonPlan(jsonPlan).await();
+
+        assertResult(
+                Arrays.asList("5," + formatSqlTimestamp(5000L), "6," + formatSqlTimestamp(6000L)),
+                sinkPath);
+    }
+
+    private static String formatSqlTimestamp(long timestamp) {
+        return TimeFormats.SQL_TIMESTAMP_FORMAT.format(unixTimestampToLocalDateTime(timestamp));
+    }
```

Comment:
Use `formatUnixTimestamp(long unixtime, TimeZone tz)` instead, as it also makes sure you use a fixed timezone and you don't get weird time zone issues when running on other machines :)

Reply:
Not sure what you mean. I can see the `formatUnixTimestamp`  only in JavaScript libraries in Flink's code base. Or is your point that the timezone should necessarily be also taken into account?

Reply:
The method I'm talking about should be in `DateTimeUtils` in table-common

Reply:
Found this addition after rebase, thanks.

Reply:
I will skip this one - the test data from the SQL suite is created based on local datetime, so the tests break if another zone is used anyhow: https://github.com/apache/flink/blob/a7192af8707f3f0d0f30fc71f3477edd92135cac/flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/utils/TestData.scala#L267
I do not see it as an issue, because 
1) No absolut values are used - only relative since 1970 for both the test data and the expected results.
2) The local time zone will this way always be used and I do not see how it can break on another machine because it using a different timezone.